requests_kerberos/kerberos_.py

```diff
@@ -108,8 +111,14 @@ def generate_request_header(self, response, host, is_preemptive=False):
 
         try:
             kerb_stage = "authGSSClientInit()"
+            # contexts still need to be stored by host, but hostname_override
+            # allows use of an arbitrary hostname for the kerberos exchange
+            # (eg, in cases of aliased hosts, internal vs external, CNAMEs
+            # w/ name-based HTTP hosting)
             result, self.context[host] = kerberos.authGSSClientInit(
-                "{0}@{1}".format(self.service, host), gssflags=gssflags)
+                "{0}@{1}".format(self.service,
+                     host if not self.hostname_override else self.hostname_override),
```

Comment:
Let's hoist this out. At this point this logical line now stretches three physical lines and involves two function calls and an inline conditional: it's just way too complex. Let's break it down slightly. =)